cluster/test-e2e.sh

```diff
@@ -21,12 +21,12 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 source "${KUBE_ROOT}/cluster/kube-util.sh"
 
 echo "Testing cluster with provider: ${KUBERNETES_PROVIDER}" 1>&2
 
-TEST_ARGS="$@"
+TEST_ARGS="$*"
```

Comment:
`TEST_ARGS` should be used in the `exec` line below

this is very similar to  https://github.com/kubernetes/kubernetes/pull/75573/files#r268014649

I'm not sure off the top of my head what the right was is to capture but then ensure spaces aren't swallowed up

I would suggest figuring out the right way and ensuring it ends up in both of these places